### PR TITLE
Optimization: DiskThreadsDiskFile::readDone copied request ptr

### PR DESCRIPTION
**Pull request by kinkie**
Detected by Coverity. CID 1529587: Unnecessary object copies can affect
performance (COPY_INSTEAD_OF_MOVE).

### CODE REVIEW

**Review by rousskov — CHANGES_REQUESTED** [review]
Please apply applicable #1797 concerns to this PR. Please do not post new PRs adding std::move wrappers until the affected PRs land.

**Review by yadij — APPROVED** [review]

**Review by rousskov — APPROVED** [review]
Thank you for this optimization. I am approving it after adjusting PR title (to make it more precise and consistent with an earlier PR) and PR description (to make it more relevant to current PR changes and more consistent with an earlier PR).

**Pull request closed without merging**
